R/deploySite.R

```diff
@@ -102,3 +91,65 @@ deploySite <- function(siteDir = getwd(),
     ...
   )
 }
+
+quartoSite <- function(path, quiet = FALSE, error_call = caller_env()) {
+  check_installed(
+    "quarto",
+    reason = "to deploy quarto sites",
+    call = error_call
+  )
+
+  config <- quarto::quarto_inspect(path)$config
+
+  list(
+    render = function() quarto::quarto_render(path, quiet = quiet, as_job = FALSE),
+    name = basename(normalizePath(path)),
+    title = config$website$title %||% config$book$title %||% config$title,
+    output_dir = outputDir(path, config$project$`output-dir`)
```

Comment:
Quarto projects that are not websites or books do not have a declared output directory; they render into the working directory.

For example, use the `_quarto.yml` alongside an `index.qmd` and `second.qmd`:

```
project:
  title: "project"
```

Running `quarto render` produces `index.html`, `second.html` and `*_files` directories for each. All are written into the working directory, not a separate output directory.

Is this configuration a "site"? It's definitely a "project". How is this project best deployed? What happens if I `render="local"`?

In `quarto::quarto_publish_site`, it looks like we publish the entire input directory, including the source files, even when we have rendered the content.

Reply:
I think of this project as a website without all the automatic treatment that comes with being a Quarto website.

I'm also OK with including the entire working directory post-rendering; mostly, I wanted you to know that this type of project has no `output-dir` in the output from `quarto inspect`.

Reply:
Oh got it; I'll add a fallback.